### PR TITLE
Fix #40 Add extras for database dependencies

### LINKED ISSUE #40: Install fails without all databases system dependencies
I tried to follow the guide, but the installation failed because locally I don't have MySQL system package dependencies and I prefer not to install them.

```pycon
$ python -m pip install -e '.[dev]'
Obtaining file:///home/paulox/Projects/django-tasks
  Installing build dependencies ... done
  Checking if build backend supports build_editable ... done
  Getting requirements to build editable ... done
  Installing backend dependencies ... done
  Preparing editable metadata (pyproject.toml) ... done
Collecting Django>=4.2 (from django-tasks==0.1.1)
  Using cached Django-5.0.6-py3-none-any.whl.metadata (4.1 kB)
Collecting typing-extensions (from django-tasks==0.1.1)
  Downloading typing_extensions-4.12.2-py3-none-any.whl.metadata (3.0 kB)
Collecting django-stubs-ext (from django-tasks==0.1.1)
  Downloading django_stubs_ext-5.0.2-py3-none-any.whl.metadata (3.5 kB)
Collecting ruff (from django-tasks==0.1.1)
  Downloading ruff-0.4.8-py3-none-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (24 kB)
Collecting coverage (from django-tasks==0.1.1)
  Using cached coverage-7.5.3-cp312-cp312-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (8.2 kB)
Collecting django-stubs[compatible-mypy] (from django-tasks==0.1.1)
  Downloading django_stubs-5.0.2-py3-none-any.whl.metadata (15 kB)
Collecting dj-database-url (from django-tasks==0.1.1)
  Downloading dj_database_url-2.2.0-py3-none-any.whl.metadata (12 kB)
Collecting psycopg (from django-tasks==0.1.1)
  Using cached psycopg-3.1.19-py3-none-any.whl.metadata (4.2 kB)
Collecting mysqlclient (from django-tasks==0.1.1)
  Downloading mysqlclient-2.2.4.tar.gz (90 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 90.4/90.4 kB 424.7 kB/s eta 0:00:00
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  error: subprocess-exited-with-error
  
  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [30 lines of output]
      /bin/sh: 1: pkg-config: not found
      /bin/sh: 1: pkg-config: not found
      /bin/sh: 1: pkg-config: not found
      Trying pkg-config --exists mysqlclient
      Command 'pkg-config --exists mysqlclient' returned non-zero exit status 127.
      Trying pkg-config --exists mariadb
      Command 'pkg-config --exists mariadb' returned non-zero exit status 127.
      Trying pkg-config --exists libmariadb
      Command 'pkg-config --exists libmariadb' returned non-zero exit status 127.
      Traceback (most recent call last):
        File "/home/paulox/.virtualenvs/django-tasks/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
          main()
        File "/home/paulox/.virtualenvs/django-tasks/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/home/paulox/.virtualenvs/django-tasks/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 118, in get_requires_for_build_wheel
          return hook(config_settings)
                 ^^^^^^^^^^^^^^^^^^^^^
        File "/tmp/pip-build-env-8fpr2zez/overlay/lib/python3.12/site-packages/setuptools/build_meta.py", line 325, in get_requires_for_build_wheel
          return self._get_build_requires(config_settings, requirements=['wheel'])
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/tmp/pip-build-env-8fpr2zez/overlay/lib/python3.12/site-packages/setuptools/build_meta.py", line 295, in _get_build_requires
          self.run_setup()
        File "/tmp/pip-build-env-8fpr2zez/overlay/lib/python3.12/site-packages/setuptools/build_meta.py", line 311, in run_setup
          exec(code, locals())
        File "<string>", line 155, in <module>
        File "<string>", line 49, in get_config_posix
        File "<string>", line 28, in find_package_name
      Exception: Can not find valid pkg-config name.
      Specify MYSQLCLIENT_CFLAGS and MYSQLCLIENT_LDFLAGS env vars manually
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× Getting requirements to build wheel did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.
```

### PR DESCRIPTION
**Pull request by pauloxnet**
Fix #40 as discussed with @RealOrangeOne during DjangoCon Europe 2024 sprints.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,9 +13,12 @@ A reference implementation and backport of background workers and tasks in Djang
 ## Installation
 
 ```
-pip install django-tasks
+python -m pip install django-tasks
 ```
 
+> [!TIP]
+> Add an extra name for each database you want to use (e.g. `django-tasks[mysql]`,  `django-tasks[postgres]` or `django-tasks[mysql,postgres]`).
+
```
Comment (RealOrangeOne) [inline README.md:21]: Nit: I don't think we should document this here. These extras are really more for dev. A user should have installed these for themselves. There's no runtime dependency on these libraries.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -14,6 +14,9 @@ source .venv/bin/activate
 python -m pip install -e '.[dev]'
 ```
 
+> [!TIP]
+> Add an extra name for each database you want to use (e.g. `[dev,mysql]`,  `[dev,postgres]` or `[dev,mysql,postgres]`).
```
Comment (RealOrangeOne) [inline CONTRIBUTING.md:18]: Suggestion: Be more explicit that installing these extras is optional
```suggestion
> Add an extra name for each database you want to develop with (e.g. `[dev,mysql]`,  `[dev,postgres]` or `[dev,mysql,postgres]`). This is optional.
```

File: README.md (hunk 1 above)
Comment (pauloxnet) [reply README.md:21]: I understand I'll remote it from README

File: CONTRIBUTING.md (hunk 2 above)
Comment (pauloxnet) [reply CONTRIBUTING.md:18]: Thanks, I'll update the PR.

**Pull request merged**